src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,596 @@
+        // create a quorum from the serviceManagerOwner
+        // get the quorum count before the quorum is created
+        uint8 quorumCountBefore = uint8(voteWeigher.quorumCount());
+        cheats.prank(serviceManagerOwner);
+        // expect each strategy to be added to the quorum
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            cheats.expectEmit(true, true, true, true, address(voteWeigher));
+            emit StrategyAddedToQuorum(quorumCountBefore, strategiesAndWeightingMultipliers[i].strategy, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+        // created quorum will have quorum number of the count before it was created
+        cheats.expectEmit(true, true, true, true, address(voteWeigher));
+        emit QuorumCreated(quorumCountBefore);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+        
+        assertEq(voteWeigher.quorumCount(), quorumCountBefore + 1);
+        // check that all of the weights are correct
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            IVoteWeigher.StrategyAndWeightingMultiplier memory strategyAndWeightingMultiplier = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumCountBefore, i);
+            assertEq(address(strategyAndWeightingMultiplier.strategy), address(strategiesAndWeightingMultipliers[i].strategy));
+            assertEq(strategyAndWeightingMultiplier.multiplier, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+    }
+
+    function testCreateQuorum_FromNotServiceManagerOwner_Reverts(
+        address notServiceManagerOwner,
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public fuzzedAddress(notServiceManagerOwner) {
+        cheats.assume(notServiceManagerOwner != serviceManagerOwner);
+        cheats.prank(notServiceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase.onlyServiceManagerOwner: caller is not the owner of the serviceManager");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+    }
+
+    function testCreateQuorum_StrategiesAndWeightingMultipliers_LengthGreaterThanMaxAllowed_Reverts(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        strategiesAndWeightingMultipliers = _removeDuplicates(strategiesAndWeightingMultipliers);
+        strategiesAndWeightingMultipliers = _replaceZeroWeights(strategiesAndWeightingMultipliers);
+
+        cheats.assume(strategiesAndWeightingMultipliers.length > voteWeigher.MAX_WEIGHING_FUNCTION_LENGTH());
+        cheats.prank(serviceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase._addStrategiesConsideredAndMultipliers: exceed MAX_WEIGHING_FUNCTION_LENGTH");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
```

Comment:
I realize the way this is implemented is that the check is in the internal function. still might be good to also make a call to `addStrategies...` and check that this reverts appropriately as well (you might need to initialize the quorum with the default list first)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/53/commits/263ce1d0bdbe1ed73daf27d600f280981d4bc262